src/skills-builder/skills-builder-modal/select-preferences/CareerInterestSelect.jsx

```diff
@@ -1,16 +1,44 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { getConfig } from '@edx/frontend-platform';
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { Stack, Row, Col } from '@edx/paragon';
+import { InstantSearch } from 'react-instantsearch-hooks-web';
+import JobTitleInstantSearch from './JobTitleInstantSearch';
+import CareerInterestCard from './CareerInterestCard';
+import { addCareerInterest } from '../../data/actions';
+import { SkillsBuilderContext } from '../../skills-builder-context';
 import messages from './messages';
 
-const CareerInterestSelect = () => (
-  <div>
-    <h4>
-      <FormattedMessage {...messages.careerInterestPrompt} />
-    </h4>
-    <p>
-      JobTitleAutosuggest component can be reused here
-    </p>
-  </div>
-);
+const CareerInterestSelect = () => {
+  const { state, dispatch, algolia } = useContext(SkillsBuilderContext);
+  const { careerInterests } = state;
+  const { searchClient } = algolia;
+
+  const handleCareerInterestSelect = (value) => {
+  // 'onSelected' function was adding a null value to the 'careerInterests' array during the onChange event
```

Comment:
Nit: I'm a little confused by the comment. I'm not sure what it's trying to tell me.

Reply:
That's a fair question — I don't think I made this very clear in the comment. It's referring to the `if` statement below that checks for a `value` to exist (as well as checking the length of the `careerInterests` array). Without checking for the value to exist, the code will dispatch a `null` value to add to the `careerInterests` array every time a user types input. So we end up with empty cards.

`onSelected` refers to the attribute of the `Form.Autosuggest` component from Paragon. I was under the impression that the code only fired when a selection was made. But it appears to fire after any `onChange` event as well. This perplexed me a bit because `Form.Autosuggest` also has a separate `onChange` attribute that we are using already.

Reply:
Ah. Gotcha. That's pretty good context. Can we try and condense the context and improve this comment?

Reply:
Yes absolutely 👍